platform-enterprise_docs/enterprise/advanced-topics/jvm-memory-tuning.md

```diff
@@ -0,0 +1,66 @@
+| `-Xms` / `-Xmx`            | Set the initial (`Xms`) and maximum (`Xmx`) heap size. The heap stores Java objects and should be 50-70% of total allocated memory. |
+| `-XX:MaxDirectMemorySize`  | Set the maximum direct (off-heap) memory. Used for NIO operations, network buffers, and file I/O.                                   |
+| `-XX:ActiveProcessorCount` | Set the number of CPUs available to the JVM. Should match the number of vCPUs allocated to the container.                           |
+
+## Resource allocation guidelines
+
+- **Heap (`-Xmx`)**: 50-70% of total allocated memory
+- **Direct memory**: 10-20% of total allocated memory
+- **Overhead** (metaspace, thread stacks, native memory): ~10% of total allocated memory
+
+Ensure total JVM memory (heap + direct memory + overhead) does not exceed container memory limits.
+
+## Example configurations
+
+The following table provides example configurations for common deployment sizes. These are starting points and may need to be tuned based on your specific usage patterns.
+
+| vCPU |  RAM  | Heap (`-Xmx`) | Direct Memory | `JAVA_OPTS`                                                                     |
+| :--: | :---: | :-----------: | :-----------: | ------------------------------------------------------------------------------- |
+|  1   | 2 GB  |     1 GB      |    512 MB     | `-XX:ActiveProcessorCount=1 -Xms500M -Xmx1000M -XX:MaxDirectMemorySize=512m`    |
+|  1   | 4 GB  |    2.5 GB     |    800 MB     | `-XX:ActiveProcessorCount=1 -Xms1000M -Xmx2500M -XX:MaxDirectMemorySize=800m`   |
+|  2   | 2 GB  |     1 GB      |    512 MB     | `-XX:ActiveProcessorCount=2 -Xms500M -Xmx1000M -XX:MaxDirectMemorySize=512m`    |
+|  2   | 4 GB  |     2 GB      |    800 MB     | `-XX:ActiveProcessorCount=2 -Xms1000M -Xmx2000M -XX:MaxDirectMemorySize=800m`   |
+|  2   | 8 GB  |     5 GB      |    1.5 GB     | `-XX:ActiveProcessorCount=2 -Xms2000M -Xmx5000M -XX:MaxDirectMemorySize=1500m`  |
+|  3   | 2 GB  |     1 GB      |    512 MB     | `-XX:ActiveProcessorCount=3 -Xms500M -Xmx1000M -XX:MaxDirectMemorySize=512m`    |
+|  3   | 4 GB  |     2 GB      |    800 MB     | `-XX:ActiveProcessorCount=3 -Xms1000M -Xmx2000M -XX:MaxDirectMemorySize=800m`   |
+|  3   | 8 GB  |     5 GB      |    1.5 GB     | `-XX:ActiveProcessorCount=3 -Xms2000M -Xmx5000M -XX:MaxDirectMemorySize=1500m`  |
+|  3   | 16 GB |     11 GB     |    2.5 GB     | `-XX:ActiveProcessorCount=3 -Xms4000M -Xmx11000M -XX:MaxDirectMemorySize=2500m` |
+
+## When to adjust memory settings
+
+Adjust your JVM memory settings if you observe the following issues in your deployment:
+
+**Increase heap memory (`-Xmx`)** if you see:
+
+- `OutOfMemoryError: Java heap space` errors in logs
+- Garbage collection pauses affecting performance
+- Steadily growing memory usage under sustained load
+
+**Increase direct memory (`MaxDirectMemorySize`)** if you see:
+
+- `OutOfMemoryError: Direct buffer memory` errors in logs
```

Comment:
Increase relative to what?
- Grant more memory at the expense of heap?
- Grant more memory at the expense of overhead?
- Something else?